lightning/src/ln/channelmanager.rs

```diff
@@ -3353,17 +3355,19 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
-	/// Provides a payment preimage in response to a PaymentReceived event, returning true and
-	/// generating message events for the net layer to claim the payment, if possible. Thus, you
-	/// should probably kick the net layer to go send messages if this returns true!
+	/// Provides a payment preimage in response to [`Event::PaymentReceived`], generating any
+	/// [`MessageSendEvent`]s needed to claim the payment.
+	///
+	/// Returns whether any HTLCs were claimed. If true, you must have the network layer process any
+	/// [`MessageSendEvent`]s returned by [`get_and_clear_pending_msg_events`].
```

Comment:
Could note that BackgroundProcessor will do this for you

Reply:
I've reworked the comment a bit to remove the part about the network layer. I feel like this is best stated elsewhere otherwise we'd need to say this on every function that could add new `MessageSendEvent`s. Seems more appropriate for higher-level documentation like on `MessageSendEvents` or `MessageSendEventsProvider`. Guess we may need to clean up the docs in this respect.  Happy to do it here or on a follow-up.

Reply:
Hmm, does seem to be an existing issue. Fine with follow-up